docs/spec_draft.md

```diff
@@ -1280,6 +1281,58 @@ For boolean element type, the behavior is same as [stablehlo.or](#stablehloor).
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.map
+
+### Semantics
+
+Performs a map function `computation` over the `inputs` along the `dimensions`
+and produces a `result` tensor where each element is the result of `computation`
+applied to the corresponding elements in `inputs`.
```

Comment:
Similarly to reduce, let's reformulate using more conventional terminology (also I think that the "where ..." part is redundant given the formalism below), e.g.

```
Applies a map function `computation` to `inputs` along the `dimensions` and 
produces a `result` tensor.
```

To distinguish this from ReduceOp, how about we rephrase ReduceOp to say:

```
Applies a reduction function `body` to `inputs` and `init_values` along 
the `dimensions` and produces a `result` tensor.
```